torch/utils/run.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+import sys
+
+def main():
+    mode = ""
+    if len(sys.argv) > 1:
+        mode = sys.argv[1]
+
+    argv_script = sys.argv.copy()
+    del argv_script[0]
+    if mode == "--inference":
```

Comment:
Sorry for not catching this earlier. It is preferable to use argparse.SubCommand instead of parsing the CLI here manually. See: https://docs.python.org/3/library/argparse.html#sub-commands

So this would be:

```
$ torchrun inference <.. args to torch.utils.inference.run...>
```

Using subcommand has many benefits like being able to print out help string for the main CLI as well as subcommands (e.g `torchrun --help` versus `torchrun inference --help`) and argparse takes care of mutual exclusivity of subcommands.

Here's an example of this in the TorchX CLI: https://github.com/pytorch/torchx/blob/main/torchx/cli/main.py#L29

There are various subcommands registered for `torchx` CLI (e.g. `$torchx run` `$torchx status` `$torchx log`)

Reply:
Done